lib/SIL/SILVerifier.cpp

```diff
@@ -4345,6 +4345,40 @@ class SILVerifier : public SILVerifierBase<SILVerifier> {
     }
     
     SILInstructionVisitor::visitSILBasicBlock(BB);
+
+    // Visit the instructions in the BB, to find holes in the lexical scopes.
+    // We use a fairly simple heuristic here (Explain heuristic).
+    llvm::DenseSet<const SILDebugScope *> AlreadySeenScopes;
```

Comment:
This probably is growing enough that can be split in its own function. I've been doing this when I was at 30k feet and I didn't want to lose this snapshot :)